src/fastoad/model_base/flight_point.py

```diff
@@ -252,19 +382,38 @@ def remove_field(cls, name):
         :param name: field name
         """
         if name in cls.__annotations__:
+            cls._redeclare_fields()
             del cls.__init__  # Delete constructor to allow it being rebuilt with dataclass() call
             delattr(cls, name)
             del cls.__annotations__[name]
             dataclass(cls)
-            if name in cls._units:
-                del cls._units[name]
 
-    def scalarize(self):
+    @classmethod
+    def _get_field_descriptors(cls) -> Dict[str, _FieldDescriptor]:
```

Comment:
Can't this be done with getters and setters ?

Reply:
Actually, it is a getter :)
But if you talk about using properties, the answer is no, because here we have a class method.